<!--
THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF
ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO
THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
PARTICULAR PURPOSE.

Copyright (c) Microsoft Corporation. All rights reserved
-->
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
    <title></title>
    <script src="/js/4_OnErrorEvent.js"></script>
</head>
<body>
    <div data-win-control="SdkSample.ScenarioInput">
        <p>
            Exceptions generated during asynchronous operations are delivered
            to user code to the optional error handler defined in the then()
            method. Note, there are a class of errors in JavaScript (for
            example, referencing an undefined variable or method) delivered
            as exceptions, but aren't something a developer would want to
            actually put in an error handler for such as syntax errors.
            <br />
            The Promise.onerror event is raised whenever an exception or
            error caught inside a promise, regardless of the presence of an
            actual error handler or not. This provides a central point where
            a developer can do logging or put a breakpoint and catch
            exceptions that would otherwise have been silently handled by the
            Promise object.
        </p>
        <button class="onErrorErrorButton action">Raise an Error</button>
        <button class="onErrorExceptionButton action">Raise an Exception</button>
    </div>
    <div data-win-control="SdkSample.ScenarioOutput">
        <div class="onErrorOutput"></div>
    </div>
</body>
</html>
